contracts/proxy/transparent/ProxyAdmin.sol

```diff
@@ -3,33 +3,26 @@
 
 pragma solidity ^0.8.19;
 
-import {ITransparentUpgradeableProxy, TransparentUpgradeableProxy} from "./TransparentUpgradeableProxy.sol";
+import {ITransparentUpgradeableProxy} from "./TransparentUpgradeableProxy.sol";
 import {Ownable} from "../../access/Ownable.sol";
+import {ShortStrings, ShortString} from "../../utils/ShortStrings.sol";
 
 /**
  * @dev This is an auxiliary contract meant to be assigned as the admin of a {TransparentUpgradeableProxy}. For an
  * explanation of why you would want to use this see the documentation for {TransparentUpgradeableProxy}.
  */
 contract ProxyAdmin is Ownable {
+    // solhint-disable-next-line var-name-mixedcase
+    ShortString public immutable PROXY_ADMIN_VERSION = ShortStrings.toShortString(string("5.0.0"));
```

Comment:
Also, using an immutable will require some processing during the constructor that a constant may not need. Since we are trying to optimise for construction cost (and not access cost) I'm dubious the short string is the way to go.

Reply:
I agree. The getter for this will probably return a `bytes32`. ShortStrings are useful with immutables that are set in the constructor. There is no reason to use an immutable here, a constant can do just as well. So it would be better to use `string public constant PROXY_ADMIN_VERSION = "5.0.0"`.